### DISCUSSION TITLE
serial update rate not a variable

### DISCUSSION THREAD

**Discussion by sstroobants** [iNavFlight/inav#7409] [General]
Hi there! 
I was looking for a way to change the update rate of the serial loop, which was possible in Betaflight using the variable serial_update_rate_hz. They added that option a couple of years ago but I see it has not been implemented in iNav yet. Is it possible to have this option here as well? Currently I just change the TASK_PERIOD_HZ of the serial task in fc_tasks.c and rebuild the firmware every time, which is also fine of course. 
Kind regards,
Stein

**Comment by DzikuVx** [comment]
Looptime is independent from serial tasks. What exactly are you trying to
do? Because I do not understand your usecase

wt., 31 sie 2021, 11:53 użytkownik sstroobants ***@***.***>
napisał:

**Reply by sstroobants** [reply]
Thanks for your reply :). My usecase is quite similar to the one described by christiaanrauch in this Issue where they discussed this for Cleanflight/Betaflight:
https://github.com/cleanflight/cleanflight/issues/2571
I want to deal with the attitude values for instance on a higher rate than 100Hz. Right now I achieve that by setting the TASK_PERIOD_HZ for the Serial task higher and rebuilding (what they basically also do with Betaflight now, but they added it as a CLI variable).

**Reply by DzikuVx** [reply]
From my perspective, this is an edge case that great majority of users will never use. If you can solve it by recompiling then good. But I don;t think we will add this setting to INAV
